src/cli/commands/run.js

```diff
@@ -11,9 +11,22 @@ import map from '../../util/map.js';
 const leven = require('leven');
 const path = require('path');
 
-// Copied from https://github.com/npm/npm/blob/63f153c743f9354376bfb9dad42bd028a320fd1f/lib/run-script.js#L175
+function quoteForCmd(arg: string): string {
+  // See the below blog post for an explanation of what's going on here:
+  // eslint-disable-next-line max-len
+  // https://blogs.msdn.microsoft.com/twistylittlepassagesallalike/2011/04/23/everyone-quotes-command-line-arguments-the-wrong-way/
+  const caretEscape = /"/.test(arg);
+  arg = `"${arg.replace(/\\+(?=$|")/g, '$&$&').replace(/"/g, '\\"')}"`;
+  if (caretEscape) {
+    arg = arg.replace(/[()%!^"<>&|]/g, '^$&');
+  }
+  return arg;
+}
+
+const quoteForShell = process.platform === 'win32' ? quoteForCmd : arg => `'${arg.replace(/'/g, "'\\''")}'`;
+
 function joinArgs(args: Array<string>): string {
-  return args.reduce((joinedArgs, arg) => joinedArgs + ' "' + arg.replace(/"/g, '\\"') + '"', '');
+  return args.length ? ' ' + args.map(quoteForShell).join(' ') : '';
```

Comment:
Yes, but the entire function would return the `' '` that gets prepended to the joined string.